packages/main/src/plugin/kubernetes-context-state.ts

```diff
@@ -227,11 +264,30 @@ class ContextsStates {
   }
 
   async dispose(name: string): Promise<void> {
-    await this.informers.get(name)?.podInformer?.stop();
-    await this.informers.get(name)?.deploymentInformer?.stop();
+    const informers = this.informers.get(name);
+    if (informers) {
+      for (const res in Object.keys(informers)) {
+        const resname = res as ResourceName;
+        await informers[resname]?.stop();
+      }
```

Comment:
wondering why we iterate on the keys and then grab values and not on the entries to avoid the `informers[resName]?`
or also why we do not simply iterate on the values ?

Reply:
For example why ContextInternalState is not a Map ?